### PR TITLE
Lighting controller port from /tg/

### PR DESCRIPTION
**Pull request by DominikPanic**
T H I C C PR ahead, nerds.

Fixes: #1992 
Fixes: #1999
Fixes: #2094 
Fixes: #2088 
Fixes: #2083 
Fixes: #2072 
Fixes: #2081 
Fixes: #2067 
Fixes: #2058 
Fixes: #1991 
Fixes: #1953 
Fixes: #2063 

:cl: LaKiller8
refactor: Lighting has been refactored.
/:cl:

### CODE REVIEW

**Comment by Rohesie** [comment]
M O O D

File: code/__HELPERS/unsorted.dm [hunk 1]
```diff
@@ -571,9 +571,9 @@ var/global/image/busy_indicator_hostile
 
 //					var/area/AR = X.loc
 
-//					if(AR.lighting_use_dynamic)							//TODO: rewrite this code so it's not messed by lighting ~Carn
+//					if(AR.dynamic_lighting)							//TODO: rewrite this code so it's not messed by lighting ~Carn
```
Comment (Rohesie) [inline code/__HELPERS/unsorted.dm]: Oh dear.

File: code/modules/lighting/lighting_corner.dm [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+/datum/lighting_corner/dummy/New()
+	return
+
+
+/datum/lighting_corner/Destroy(force)
+	if(!force)
+		return QDEL_HINT_LETMELIVE
+
+	stack_trace("Ok, Look, /tg/, I need you to find whatever fucker decided to call qdel on a fucking lighting corner, then tell him very nicely and politely that he is 100% retarded and needs his head checked. Thanks. Send them my regards by the way.")
```
Comment (Rohesie) [inline code/modules/lighting/lighting_corner.dm]:
```suggestion
	stack_trace("Ok, Look, /tg/, I need you to find whatever fucker decided to call qdel on a fucking lighting corner, then tell him very nicely and politely that he is 100% retarded and needs his head checked. Thanks. Send them my regards by the way.") // PJB3005
```
Credit where it's due.

File: code/__HELPERS/unsorted.dm (hunk 1 above)
Comment (DominikPanic) [reply code/__HELPERS/unsorted.dm]: m a s s r e p l a c e

Comment (Rohesie) [reply code/__HELPERS/unsorted.dm]: ~Carn

**Comment by spookydonut** [comment]
needs optimising to avoid lag

**Comment by Rohesie** [comment]
New conflict.

File: code/modules/mob/living/carbon/carbon.dm [hunk 3]
```diff
@@ -381,4 +381,42 @@
...
+		return
+
+	sight = initial(sight)
+	lighting_alpha = initial(lighting_alpha)
+	see_in_dark = species.darksight
+	see_invisible = initial(see_invisible)
+
+	if(client.eye != src)
+		var/atom/A = client.eye
+		if(A.update_remote_sight(src)) //returns 1 if we override all other sight updates.
```
Comment (psykzz) [inline code/modules/mob/living/carbon/carbon.dm]:
```suggestion
		if(A?.update_remote_sight(src)) //returns 1 if we override all other sight updates.
```

Comment (psykzz) [reply code/modules/mob/living/carbon/carbon.dm]:
```
[21:25:47] Runtime in carbon.dm, line 405: Cannot execute null.update remote sight().
proc name: update sight (/mob/living/carbon/human/update_sight)
usr: Zyzara/(Oliver Kifer)
src: Oliver Kifer (/mob/living/carbon/human)
src.loc: null
call stack:
Oliver Kifer (/mob/living/carbon/human): update sight()
Oliver Kifer (/mob/living/carbon/human): reset perspective(null)
Oliver Kifer (/mob/living/carbon/human): Login()
Oliver Kifer (/mob/living/carbon/human): Login()
Oliver Kifer (/mob/living/carbon/human): Login()
/datum/mind (/datum/mind): transfer to(Oliver Kifer (/mob/living/carbon/human), 1)
Distress Signal (/datum/game_mode/distress): end of round deathmatch()
Distress Signal (/datum/game_mode/distress): declare completion(0)
Distress Signal (/datum/game_mode/distress): declare completion(0)
Ticker (/datum/controller/subsystem/ticker): fire(0)
Ticker (/datum/controller/subsystem/ticker): ignite(0)
```

Comment (DominikPanic) [reply code/modules/mob/living/carbon/carbon.dm]: That would be supressing a runtime. Client eye should not be null.

**Pull request merged**